test/pop_protocol_test.erl

```diff
@@ -6,12 +6,34 @@
 
 -include("../src/potato_records.hrl").
 
+make_transaction(PrivateKey, PublicKey, Nonce, ChainId) ->
+    T0 = #{
+	   game_data => undefined,
+	   nonce => Nonce,
+	   player_id => PublicKey,
+	   consensus_data => #{
+			       signature => undefined,
+			       chain_id => ChainId
+			      }
+	  },
+    Hash = my_crypto:hash(my_serializer:serialize_object(T0)),
+    Signature = my_crypto:sign(Hash, PrivateKey),
+    CD = maps:get(consensus_data, T0),
```

Comment:
DDD:

Funny, I think either in Ethereum they do the same thing where the sig is of the whole object with a nil signature field rather than signing the object and adding the signature + object in some new struct.